OpenRA.Mods.Common/Traits/Render/WithSpriteBody.cs

```diff
@@ -61,8 +64,12 @@ protected WithSpriteBody(ActorInitializer init, WithSpriteBodyInfo info, Func<in
 			Func<bool> paused = () => IsTraitPaused &&
 				DefaultAnimation.CurrentSequence.Name == NormalizeSequence(init.Self, Info.Sequence);
 
+			Func<WVec> subtractDAT = () => info.ForceToGround ? 
```

Comment:
It would be better to do this as:
```csharp
Func<WVec> subtractDAT = null;
if (info.ForceToGround)
	subtractDAT = () => new WVec(0, 0, -init.Self.World.Map.DistanceAboveTerrain(init.Self.CenterPosition).Length)
```

The current code forces a function evaluation for all cases, even when it returns nothing useful.

Reply:
Done.